packages/web-components/src/text-input/text-input.spec.ts

```diff
@@ -0,0 +1,636 @@
+      node.appearance = 'outline';
+    });
+
+    await expect(element).toHaveAttribute('appearance', 'outline');
+    await expect(element).toHaveJSProperty('appearance', 'outline');
+
+    await element.evaluate((node: TextInput) => {
+      node.appearance = 'underline';
+    });
+
+    await expect(element).toHaveAttribute('appearance', 'underline');
+    await expect(element).toHaveJSProperty('appearance', 'underline');
+
+    await element.evaluate((node: TextInput) => {
+      node.appearance = 'filled-darker';
+    });
+    await expect(element).toHaveAttribute('appearance', 'filled-darker');
+    await expect(element).toHaveJSProperty('appearance', 'filled-darker');
+
+    await element.evaluate((node: TextInput) => {
+      node.appearance = 'filled-lighter';
+    });
+    await expect(element).toHaveAttribute('appearance', 'filled-lighter');
+    await expect(element).toHaveJSProperty('appearance', 'filled-lighter');
+  });
+
+  test('should initialize to the `initialValue` property if no value property is set', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+                <fluent-text-input></fluent-text-input>
+            `;
+    });
+
+    const initialValue = await element.evaluate<string, TextInput>(node => node.initialValue);
+
+    await expect(element).toHaveJSProperty('value', initialValue);
+  });
+
+  test('should initialize to the provided `value` attribute if set pre-connection', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+                <fluent-text-input value="foo"></fluent-text-input>
+            `;
+    });
+
+    const element = page.locator('fluent-text-input');
+
+    await expect(element).toHaveJSProperty('value', 'foo');
+  });
+
+  test('should initialize to the provided `value` property if set pre-connection', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ ``;
+
+      const textField = document.createElement('fluent-text-input') as TextInput;
+
+      textField.value = 'foo';
+
+      node.appendChild(textField);
+    });
+
+    await expect(element).toHaveJSProperty('value', 'foo');
+  });
+
+  test('should initialize to the provided `value` attribute if set post-connection', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+                <fluent-text-input></fluent-text-input>
+            `;
+    });
+
+    await element.evaluate(node => {
+      node.setAttribute('value', 'foo');
+    });
+
+    await expect(element).toHaveJSProperty('value', 'foo');
+  });
+
+  test('should hide the label when no default slotted content is provided', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = '<fluent-text-input></fluent-text-input>';
+    });
+
+    await expect(label).toHaveClass(/label__hidden/);
+  });
+
+  test('should hide the label when start content is provided', async () => {
```

Comment:
![image](https://github.com/microsoft/fluentui/assets/12279161/90d4dd20-2c7c-44d3-a3a9-1c97da7f8074)

Reply:
In our story, there is start content but the label is still visible

Reply:
That's only because the story has an explicit label passed through the default slot. If you remove <fluent-label> from the story it renders correctly..

![image](https://github.com/microsoft/fluentui/assets/11062709/fb82fbfb-e5a0-4c8e-99b2-5599655314e0)